dlpy/attribute_utils.py

```diff
@@ -55,6 +55,14 @@ class :class:`DataSpec` or the resulting attribute table will
         
     '''
 
+    # ensure list of strings
+    if not isinstance(layers,list):
+        raise TypeError('layers must be a list of strings.')
+
+    # ensure list of data specs
+    if not isinstance(data_spec,list):
```

Comment:
You are correct, the test simply checks to see whether the input is a list, which was my real concern.  I can either make the check more robust or change the comment.  Perhaps it would be best to make the check more robust.

Reply:
I just updated the error-checking for both the layers list and the data_spec list.  I think this should resolve the problem you pointed out.